### PR TITLE
improving hi/lo id generation

### PR DESCRIPTION
**Pull request by gavinking**
Don't issue concurrent requests for new hi values.

see #411

@Sanne @cescoffier  does this look right to you guys?

### CODE REVIEW

File: hibernate-reactive-core/src/main/java/org/hibernate/reactive/id/impl/TableReactiveIdentifierGenerator.java [hunk 1]
```diff
@@ -89,7 +94,42 @@ private synchronized long next(long hi) {
...
 		}
+		else {
+			synchronized (this) {
+				CompletableFuture<Long> result = new CompletableFuture<>();
+				if (queue == null) {
+					// make a queue for any concurrent streams
+					queue = new ArrayList<>();
+					// go off and fetch the next hi value from db
+					nextHiValue(session).thenAccept( id -> {
+//						Vertx.currentContext().runOnContext(v -> {
```
Comment (gavinking) [inline hibernate-reactive-core/src/main/java/org/hibernate/reactive/id/impl/TableReactiveIdentifierGenerator.java:105]: I'm not sure if this is needed or not.

**Review by cescoffier — APPROVED** [review]
The pattern looks good.

**Comment by gavinking** [comment]
@Sanne if you don't respond I'm going to just merge it ;-)

**Comment by DavideD** [comment]
He is on PTO, today.
I suppose we can merge it. We already have a test somewhere for this isn't it?

**Comment by DavideD** [comment]
> We already have a test somewhere for this isn't it?

Well, I suppose lot of tests will fail if this doesn't work

**Comment by gavinking** [comment]
> He is on PTO, today.
> I suppose we can merge it. We already have a test somewhere for this isn't it?

Yeah, and I know something useful is being tested there, because I had to struggle more than I expected to get the tests to pass when making this change. (I had not originally considered that there could be more queued id requests than the block size.)

What we *don't* have is any tests for "concurrent" streams. But I'm totally not going to try adding them right now.

**Comment by DavideD** [comment]
Sounds good to me

**Pull request merged**